hal/src/electron/modem/mdm_hal.h

```diff
@@ -661,6 +665,11 @@ class MDMParser
     volatile bool _cancel_all_operations;
     volatile uint32_t _error;
     system_tick_t _lastProcess;
+    system_tick_t _lastVerboseC_regUpdate;
+    bool _verboseCregEnabled;
+    bool _verboseCgregEnabled;
+    bool _verboseCeregEnabled;
+    int _reinitModem;
```

Comment:
May be Nit: It looks like `_reinitModem` takes these two values `0xFF` and `0x55` and no operations are being done other than assigning and checking. May be can get away with `bool` instead of `int`?

Reply:
I made this toggle multiple bits because it's a sensitive operation (re-init modem).  So it's at least somewhat tolerant to cosmic rays toggling bits :D

Reply:
It's not a joke... plenty of critical operations use multi-bit/magic numbers.

Reply:
We have a lot of other critical pieces of code, why only here specifically? And I would argue if we are worried about something flipping bits in SRAM, we have bigger problems on our hands.

Reply:
I would prefer to have a few extra bits in between power cycling the modem and normal operation.  I think it's a reasonable thing to add.

Reply:
@technobly Let's not overcomplicate things. These magic numbers don't protect us against anything. If we are that worried about errors in SRAM, going by the same logic the same thing may happen:
1. In register that holds the value you are comparing with
2. While a constant is being loaded from flash into cache
3. In cache
4. etc etc etc

Reply:
I ended up implementing your suggestion of using `_error` and made the code a bit more safe by changing the number of `_error`'s that would cause a modem reinit from a magic number of 2, to a define.